### ISSUE TITLE
Nondeterminstic login

### ISSUE THREAD

**Issue by djrenren** [chezbob/chezbob#20]
Users are reporting that they scan their ID, sign in as user X, sign out, scan the same ID, sign in as user Y.

This should be impossible because barcodes are unique in the database, so the barcode must be registered one of them. Likely the issue is either in scanning from the `nfc` reader or the card has randomized their ID and created a conflict. Given that we've seen this happen twice already, that would be statistically unlikely (if ID cycling is random, which I guess isn't guaranteed).

**Comment by djrenren** [comment]
So it looks like an NFC reading problem:

Read 1:
```
Apr 12 12:34:01 bobolith env[411730]:   header: {
Apr 12 12:34:01 bobolith env[411730]:     id: '00b5c8ba-ebbd-41be-a7dd-6843446de752',
Apr 12 12:34:01 bobolith env[411730]:     to: 'pos',
Apr 12 12:34:01 bobolith env[411730]:     type: 'scan_event',
Apr 12 12:34:01 bobolith env[411730]:     from: 'nfc'
Apr 12 12:34:01 bobolith env[411730]:   },
Apr 12 12:34:01 bobolith env[411730]:   body: { barcode: '63' }
```


Read 2:
```
Apr 12 12:34:13 bobolith env[411730]:   header: {
Apr 12 12:34:13 bobolith env[411730]:     id: '65ebd90d-5c1b-4344-b539-728c4302a8d2',
Apr 12 12:34:13 bobolith env[411730]:     to: 'pos',
Apr 12 12:34:13 bobolith env[411730]:     type: 'scan_event',
Apr 12 12:34:13 bobolith env[411730]:     from: 'nfc'
Apr 12 12:34:13 bobolith env[411730]:   },
Apr 12 12:34:13 bobolith env[411730]:   body: { barcode: '047e274a975e8090' }
```

**Comment by djrenren** [comment]
Okay this is a really funny bug. We just kinda banged on the NFC reader until we got consistent results, but uhhh we missed an important part of the spec:

See "response data": https://en.wikipedia.org/wiki/Smart_card_application_protocol_data_unit

Now look at our code:

https://github.com/chezbob/chezbob/blob/271c48c7cfdda140eba60a63ae9a69b087700d60/devices/nfc/nfc.js#L43-L64

There are actually 2 problems with this code:

1. We tell the NFC reader that we will accept any number of bytes in return, but we only allocate 40 bytes for the response
2. We don't check the status code! In fact, we include the status code in the barcodes!

This means that one student has the error status code `63` as their barcode, and all other students have their actual ID + `90`.

## The fix
- Inspect the final two bytes of the response to look for the status code. If it's not `0x9000` throw an error.

This is more of an ops thing but we should remove the registration of the error barcode. For now,  We should just leave the `90` suffix in place. If we want to remove that later, we can, we'll just have to write a database migration, so I'll open a new issue for that.

**Comment by alex-bellon** [comment]
Closed by #41
